spinn_front_end_common/interface/front_end_common_interface_functions.py

```diff
@@ -429,10 +430,11 @@ def wait_for_cores_to_be_ready(
                                                            CPUState.C_MAIN)
         # check that everything has gone though c main to reach correct sync or
         # failing for some unknown reason
-        while processor_c_main != 0:
-            time.sleep(0.1)
-            processor_c_main = self._txrx.get_core_state_count(app_id,
-                                                               CPUState.C_MAIN)
+        while True:
```

Comment:
My issue is mainly waiting for 0.1 seconds. We only _run_ for 0.1 seconds and if the first check fails by 0.01 seconds, we waste a lot of time relative to our simulation time.

You're right, it could be solved without `while True`. My main concern is the wait time which is not appropriate to all use cases.